policies/document_stream_insertion.md

```diff
@@ -0,0 +1,54 @@
+# Document Stream Insertion Policy (Dynamic Mark-up)
+
+
+## Objective
+The API around document stream modification ([dynamic mark-up insertion](https://www.w3.org/TR/2011/WD-html5-author-20110705/apis-in-html-documents.html#dynamic-markup-insertion)), i.e., `document.write`, `document.writeln`, `document.open` and `document.close` 
+are anti-pattern, parser-blocking javascritp API. The objective of the proposed feature is to limit the usage of this API in websites.
+
+## Solution
+
+When the feature is disabled for a specific origin in a frame, any occurance of such API will be ignored and might
+lead to a DOMException<sup>[1](#notes)</sup>.
+
+## Usage
+The feature can be set through the HTTP headers in the response. For instance
+
+```
+Feature policy: document-stream-insertion 'none'
+```
+
+which would disable the feature for all same and cross origin content. The feature can also be set for a given `<iframe>`
+by modifying the `allow` attribute. For instance,
+
+```
+<iframe src="https://example.com" allow="document-stream-insertion http://www.google.com"></iframe>
+```
+would limit the usage of stream insertion API to only "google.com" origins using HTTPS protocol.
```

Comment:
Not quite accurate: This limits it to just www.google.com, not any *.google.com origin. (and you've used HTTP, rather than HTTPS)

Reply:
Thanks, I believe I confused site instance with origin.